hail/python/test/hailtop/batch/test_batch.py

```diff
@@ -521,3 +542,24 @@ def test_envvar(self):
         j.env('SOME_VARIABLE', '123abcdef')
         j.command('[ $SOME_VARIABLE = "123abcdef" ]')
         assert b.run().status()['state'] == 'success'
+
+    def test_single_job_with_shell(self):
+        with tempfile.NamedTemporaryFile('w') as output_file:
+            msg = 'hello world'
+
+            b = self.batch()
+            j = b.new_job(shell='bin/sh')
+            j.command(f'echo "{msg}" > {j.ofile}')
+
+            b.write_output(j.ofile, output_file.name)
+            b.run(wait=True)
+
+            assert self.read(output_file.name) == msg
+
+    def test_single_job_with_nonsense_shell(self):
+        msg = 'hello world'
+
+        b = self.batch()
+        j = b.new_job(shell='bin/ajdsfoijasidojf')
+        j.command(f'echo "{msg}"')
+        assert b.run().status()['state'] != 'success'
```

Comment:
You run this command twice in two different tests.  This is not the fastest, and we've consolidated multiple asserts in other batch tests.

Reply:
Consolidated as suggested, and left in to test the service backend handling of shell.

Reply:
Oops, sorry, I misread the diff, this was fine.

Reply:
I think you want the more specific `== 'failure'` here.  It would be incorrect if the state was, `cancelled`, say.